src/system.c

```diff
@@ -383,8 +383,8 @@ void HAL_SPI_TxRxCpltCallback(SPI_HandleTypeDef *hspi)
   struct complete_packet *rx_pkt = (struct complete_packet *)RX_Buffer;
 
   /* Limit the amount of data copied to prevent buffer overflow. */
-  if (rx_pkt->header.size > sizeof(rx_pkt_userspace))
-    rx_pkt->header.size = sizeof(rx_pkt_userspace);
+  if (rx_pkt->header.size > SPI_DMA_BUFFER_SIZE)
```

Comment:
What's wrong with `sizeof(rx_pkt_userspace)` that's not wrong in the `sizeof(msg)` above? :thinking:

Reply:
`rx_pkt_userspace` is a pointer, not an array :sweat_smile: So its sizeof is 4 bytes

Reply:
Damnit, that's a good one :fire: